### DISCUSSION TITLE
Going "unavailable" for 30 seconds every 20 seconds

### DISCUSSION THREAD

**Discussion by TheRydad** [jlvaillant/intellicenter#38] [Q&A]
First off, thank you for this integration! I am really excited to get it setup.

I just got it installed. The integration discovered my installation immediately. When I first went to the device in homeassistant, all controls and sensors were there and appeared to be working perfectly. But then... they blinked off. 

While was scratching my and trying to figure out what to do next, they came back on. Then off. The on again. Looking at the history of one of the lights, it is perfectly consistent with showing status for 20 seconds, then going unavailable for 30 and repeating that cycle.  seems like it is 20/30 seconds exactly.

Any thoughts on where I should look to troubleshoot?

A few notes:
-The Intellicenter and HA are on different VLANS. I believe I have all of the mDNS and Bonjour routing and reflectors setup correctly. I am able to consistently see the Intellicenter as a "local" pool when I am using the iOS app locally.
-The initial discovery found everything perfectly.
-I am not opposed to putting it all on the same VLAN.
-If it is potentially related to the discovery protocol, is there a static configuration I can try?

I really appreciate any pointers. And thanks again for putting this together! I was surprised it wasn't one of the "built-in" HA integrations.

EDIT: I realize I should have mentioned I was already on the 1.064 release when I installed the integration.
EDIT2: I now have HA and Intellicenter on same VLAN, but still disconnecting. Logfile excerpt below.

Logfile excerpt (I included a few repeats because I wasn't sure what might be leading what):
2022-02-08 17:23:57 WARNING (MainThread) [custom_components.intellicenter.pyintellicenter.controller] CONTROLLER: error 400 : {'command': 'Error', 'messageID': '2decd9da-06bc-4d87-b832-25fb4bb091c7', 'response': '400', 'description': "ParseError: expected value, got 'p' (112)"}
2022-02-08 17:24:07 ERROR (MainThread) [custom_components.intellicenter.pyintellicenter.protocol] PROTOCOL: heartbeat missed, closing connection
2022-02-08 17:24:07 ERROR (MainThread) [custom_components.intellicenter.pyintellicenter.controller] system disconnected  from 10.x.x.x 
2022-02-08 17:24:07 INFO (MainThread) [custom_components.intellicenter.pyintellicenter.controller] will attempt to reconnect in 30s
2022-02-08 17:24:37 INFO (MainThread) [custom_components.intellicenter.pyintellicenter.controller] model now contains 47 objects
2022-02-08 17:24:47 WARNING (MainThread) [custom_components.intellicenter.pyintellicenter.controller] CONTROLLER: error 400 : {'command': 'Error', 'messageID': '8c4bd1fe-07fb-47cb-a993-9e012ef7eb86', 'response': '400', 'description': "ParseError: expected value, got 'p' (112)"}
2022-02-08 17:24:57 ERROR (MainThread) [custom_components.intellicenter.pyintellicenter.protocol] PROTOCOL: heartbeat missed, closing connection
2022-02-08 17:24:57 ERROR (MainThread) [custom_components.intellicenter.pyintellicenter.controller] system disconnected  from 10.x.x.x 
2022-02-08 17:24:57 INFO (MainThread) [custom_components.intellicenter.pyintellicenter.controller] will attempt to reconnect in 30s
2022-02-08 17:25:27 INFO (MainThread) [custom_components.intellicenter.pyintellicenter.controller] model now contains 47 objects

**Comment by Jizzay1** [comment]
> First off, thank you for this integration! I am really excited to get it setup.
> 
> I just got it installed. The integration discovered my installation immediately. When I first went to the device in homeassistant, all controls and sensors were there and appeared to be working perfectly. But then... they blinked off.
> 
> While was scratching my and trying to figure out what to do next, they came back on. Then off. The on again. Looking at the history of one of the lights, it is perfectly consistent with showing status for 20 seconds, then going unavailable for 30 and repeating that cycle. seems like it is 20/30 seconds exactly.
> 
> Any thoughts on where I should look to troubleshoot?
> 
> A few notes: -The Intellicenter and HA are on different VLANS. I believe I have all of the mDNS and Bonjour routing and reflectors setup correctly. I am able to consistently see the Intellicenter as a "local" pool when I am using the iOS app locally. -The initial discovery found everything perfectly. -I am not opposed to putting it all on the same VLAN. -If it is potentially related to the discovery protocol, is there a static configuration I can try?
> 
> I really appreciate any pointers. And thanks again for putting this together! I was surprised it wasn't one of the "built-in" HA integrations.
> 
> EDIT: I realize I should have mentioned I was already on the 1.064 release when I installed the integration. EDIT2: I now have HA and Intellicenter on same VLAN, but still disconnecting. Logfile excerpt below.
> 
> Logfile excerpt (I included a few repeats because I wasn't sure what might be leading what): 2022-02-08 17:23:57 WARNING (MainThread) [custom_components.intellicenter.pyintellicenter.controller] CONTROLLER: error 400 : {'command': 'Error', 'messageID': '2decd9da-06bc-4d87-b832-25fb4bb091c7', 'response': '400', 'description': "ParseError: expected value, got 'p' (112)"} 2022-02-08 17:24:07 ERROR (MainThread) [custom_components.intellicenter.pyintellicenter.protocol] PROTOCOL: heartbeat missed, closing connection 2022-02-08 17:24:07 ERROR (MainThread) [custom_components.intellicenter.pyintellicenter.controller] system disconnected from 10.x.x.x 2022-02-08 17:24:07 INFO (MainThread) [custom_components.intellicenter.pyintellicenter.controller] will attempt to reconnect in 30s 2022-02-08 17:24:37 INFO (MainThread) [custom_components.intellicenter.pyintellicenter.controller] model now contains 47 objects 2022-02-08 17:24:47 WARNING (MainThread) [custom_components.intellicenter.pyintellicenter.controller] CONTROLLER: error 400 : {'command': 'Error', 'messageID': '8c4bd1fe-07fb-47cb-a993-9e012ef7eb86', 'response': '400', 'description': "ParseError: expected value, got 'p' (112)"} 2022-02-08 17:24:57 ERROR (MainThread) [custom_components.intellicenter.pyintellicenter.protocol] PROTOCOL: heartbeat missed, closing connection 2022-02-08 17:24:57 ERROR (MainThread) [custom_components.intellicenter.pyintellicenter.controller] system disconnected from 10.x.x.x 2022-02-08 17:24:57 INFO (MainThread) [custom_components.intellicenter.pyintellicenter.controller] will attempt to reconnect in 30s 2022-02-08 17:25:27 INFO (MainThread) [custom_components.intellicenter.pyintellicenter.controller] model now contains 47 objects

I can also confirm this behavior after upgrading from the prior firmware. The IntelliCenter is on the same vlan as HA. Thankas for looking into this - this is an awesome integration!

**Comment by sellison2** [comment]
I have confirmed this is fixed in v1.1.0
